docs/docs/dev_docs/contracts/syntax/storage/storage_slots.md

```diff
@@ -0,0 +1,63 @@
+---
+title: Storage slots
+---
+
+From the description of [storage slot in concepts](./../../../../concepts/foundation/state_model/storage_slots.md) you will get an idea around the logic of storage slots. In this section we will go into more detail to walk through an entire example of how storage slots are computed for private state to improve that intuition. Recall, that storage slots are only a logical construct in private, and are not "actually" used for lookups, but rather as a value to constrain against.
+
+For the case of the example, we will look at what is inserted into the private data tree when adding a note in the Token contract. Specifically, we are looking at the last part of the `transfer` function:
+
+#include_code increase_private_balance yarn-project/noir-contracts/src/contracts/token_contract/src/main.nr rust
+
+This function is creating a new note and inserting it note into the balance set of the recipient `to`. Recall that to ensure privacy, only the note commitment is really inserted into the private data tree. To share the contents of the note with `to` the contract can emit an encrypted log (which this one does), or it can require an out-of-band data transfer sharing the information. Below, we will walk through the steps of how the note commitment is computed and inserted into the tree. For this, we don't care about the encrypted log, so we are going to ignore that part of the function call for now. 
```

Comment:
```suggestion
This function is creating a new note and inserting it into the balance set of the recipient `to`. Recall that to ensure privacy, only the note commitment is really inserted into the private data tree. To share the contents of the note with `to` the contract can emit an encrypted log (which this one does), or it can require an out-of-band data transfer sharing the information. Below, we will walk through the steps of how the note commitment is computed and inserted into the tree. For this, we don't care about the encrypted log, so we are going to ignore that part of the function call for now. 
```